phist@1.11.2 fails build on ARM64: kernels/builtin/axpy_kernels_nt.c:22:10: fatal error: emmintrin.h: No such file or directory

### Steps to reproduce the issue

`phist@1.11.2` build fails using:
* `spack@develop` (09e1258ed42f00ea9bde9857f233eeda5c61a895 from `Sat Jul 8 15:36:27 2023 -0500`)
* GCC 11.1.0
* Ubuntu 20.04, ARM64

Spack environment: [spack.yaml.txt](https://github.com/spack/spack/files/11992642/spack.yaml.txt)

<details><summary>Concretization</summary>
<pre>
 -   5bv4pof  phist@1.11.2%gcc@11.1.0+fortran+host+int64~ipo+mpi+openmp~parmetis+scamac+shared~trilinos build_system=cmake build_type=Release generator=make kernel_lib=builtin outlev=2 patches=1fc9ac6 arch=linux-ubuntu20.04-aarch64
[+]  cib64fo      ^cmake@3.26.3%gcc@11.1.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-ubuntu20.04-aarch64
[+]  tq4s2d2          ^ncurses@6.3%gcc@11.1.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  2yo52cd          ^openssl@1.1.1u%gcc@11.1.0~docs~shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-aarch64
[+]  esd7eb6              ^ca-certificates-mozilla@2023-01-10%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-aarch64
[+]  6iddyph      ^gmake@4.4.1%gcc@11.1.0~guile build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  67iy4j5          ^gnuconfig@2022-09-17%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-aarch64
[+]  la56q6q      ^mpich@4.1.1%gcc@11.1.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs~wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-aarch64
[+]  jyhftkv          ^findutils@4.9.0%gcc@11.1.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-aarch64
[+]  wr2tvqh          ^libfabric@1.18.1%gcc@11.1.0~debug~kdreg build_system=autotools fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu20.04-aarch64
[+]  rn4yory          ^libpciaccess@0.17%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  jkbd2o4              ^libtool@2.4.7%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  ywapq5v              ^util-macros@1.19.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  k5yt2z5          ^libxml2@2.10.3%gcc@11.1.0~python build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  hrknc7y              ^libiconv@1.17%gcc@11.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-aarch64
[+]  352xzo3          ^pkgconf@1.9.5%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  g33gvtp          ^yaksa@0.2%gcc@11.1.0~cuda~rocm build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  qbsn4gd              ^autoconf@2.69%gcc@11.1.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-aarch64
[+]  2e2so7h              ^automake@1.16.5%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  uhlwdot              ^m4@1.4.19%gcc@11.1.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-aarch64
[+]  gbmzhic                  ^libsigsegv@2.14%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  apntqpp      ^openblas@0.3.23%gcc@11.1.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=openmp arch=linux-ubuntu20.04-aarch64
[+]  x52hmjj          ^perl@5.38.0%gcc@11.1.0+cpanm+open+shared+threads build_system=generic arch=linux-ubuntu20.04-aarch64
[+]  rke2xrc              ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-aarch64
[+]  5crj7v2      ^python@3.10.12%gcc@11.1.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-ubuntu20.04-aarch64
[+]  hcntjnj          ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-aarch64
[+]  xx5z4xk              ^diffutils@3.9%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  ljb5exf          ^expat@2.5.0%gcc@11.1.0+libbsd build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  euclh35              ^libbsd@0.11.7%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  oms2klk                  ^libmd@1.0.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  qhmcs6x          ^gdbm@1.23%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  vy4wcv5          ^gettext@0.21.1%gcc@11.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  fxfhemw              ^tar@1.34%gcc@11.1.0 build_system=autotools zip=pigz arch=linux-ubuntu20.04-aarch64
[+]  h3c7ba6                  ^pigz@2.7%gcc@11.1.0 build_system=makefile arch=linux-ubuntu20.04-aarch64
[+]  dzvxsh7                  ^zstd@1.5.5%gcc@11.1.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-aarch64
[+]  amu3day          ^libffi@3.4.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  rqy5l5i          ^libxcrypt@4.4.35%gcc@11.1.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu20.04-aarch64
[+]  rugrrau          ^readline@8.2%gcc@11.1.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-aarch64
[+]  nxqd2ih          ^sqlite@3.42.0%gcc@11.1.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  2jofsst          ^util-linux-uuid@2.38.1%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]  izauomq          ^xz@5.4.1%gcc@11.1.0+pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-aarch64
[+]  c7nojld          ^zlib@1.2.13%gcc@11.1.0+optimize+pic+shared build_system=makefile arch=linux-ubuntu20.04-aarch64
</pre>
</details>

Error:
```
$> spack -e . install
...
==> Installing phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb
==> No binary for phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/e2/e23f76307c26b930f7331a734b0a864ea6d7fb4a13c12f3c5d70c2c41481747b.tar.gz
==> Applied patch /spack/var/spack/repos/builtin/packages/phist/sbang.patch
==> Ran patch() for phist
==> phist: Executing phase: 'cmake'
==> phist: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/gmake-4.4.1-6iddypha2yhaos2uprfaop3rfbclkcf7/bin/make' '-j32'

8 errors found in build log:
     444     [ 21%] Generating precon_d/precon_d.h
     445     cd /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/fortran_bindings && sed s/"^#.*"//g /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-buil
             d-5bv4pof/fortran_bindings/precon_d/precon_d-2.CFWRAPPER_SKIP.h > /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/fortran_bindings/precon_d/precon_d.h
     446     [ 22%] Generating precon_c/precon_c.h
     447     cd /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/fortran_bindings && sed s/"^#.*"//g /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-buil
             d-5bv4pof/fortran_bindings/precon_c/precon_c-2.CFWRAPPER_SKIP.h > /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/fortran_bindings/precon_c/precon_c.h
     448     [ 22%] Generating phist_kernels.F90, phist_jada.F90, phist_tools.F90, phist_tools_s.F90, phist_tools_d.F90, phist_tools_c.F90, phist_tools_z.F90, phist_kernels_s.F90, phist_kernels_d.F90, phist_kernels_c.F90, phist_kernels_z.F90,
              phist_core_s.F90, phist_core_d.F90, phist_core_c.F90, phist_core_z.F90, phist_krylov_s.F90, phist_krylov_d.F90, phist_krylov_c.F90, phist_krylov_z.F90, phist_jada_s.F90, phist_jada_d.F90, phist_jada_c.F90, phist_jada_z.F90, phis
             t_precon_s.F90, phist_precon_d.F90, phist_precon_c.F90, phist_precon_z.F90
     449     cd /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/fortran_bindings && ./cfwrapper.py
  >> 450     /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/fortran_bindings/./cfwrapper.py:268: DeprecationWarning: 'U' mode is deprecated
     451       whole_file = open(directory[0] + "/" + c_file_name, 'rU',
  >> 452     /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/fortran_bindings/./cfwrapper.py:342: DeprecationWarning: 'U' mode is deprecated
     453       whole_file_original = open(directory[0] + "/" + c_file_name, 'rU',
     454
     455     Current directory: /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/fortran_bindings
     456
     457     Pass 1: looking for enumerators, funptr and derived types...
     458     Pass 2: looking for C functions...

     ...

     1153    /spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/mpich-4.1.1-la56q6qmhnlgtwksq4vpn6mnewxxgzg3/bin/mpic++ -Dphist_kernels_builtin_EXPORTS -I/spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/openblas-0.3.23-apntqppplgorwh6
             qpr3dp5ny3pox3axi/include -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/drivers/matfuncs -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/dr
             ivers/matfuncs/scamac -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/drivers/matfuncs/scamac -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb
             /spack-build-5bv4pof -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels/
             builtin -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels/trilinos_interface -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/to
             ols -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/core -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/krylov -I/tmp/root/spack-sta
             ge/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/jada -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/precon -I/tmp/root/spack-stage/spack-stage-phist-1.11
             .2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/drivers -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/cxx_bindings -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacag
             cksw3uh3wrx4lezb56myb/spack-src/lib/googletest-master-12-2015 -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src -march=native     -fopenmp -O3 -fno-math-errno -ffinite-math-only -fno-
             signed-zeros -fno-trapping-math -std=gnu++11 -fPIC -MD -MT CMakeFiles/phist_kernels_builtin.dir/src/kernels/common/kernels_common_impl.cpp.o -MF CMakeFiles/phist_kernels_builtin.dir/src/kernels/common/kernels_common_impl.cpp.o.d
             -o CMakeFiles/phist_kernels_builtin.dir/src/kernels/common/kernels_common_impl.cpp.o -c /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels/common/kernels_common_impl.cpp
     1154    /spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/mpich-4.1.1-la56q6qmhnlgtwksq4vpn6mnewxxgzg3/bin/mpicc -Dphist_kernels_builtin_EXPORTS -I/spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/openblas-0.3.23-apntqppplgorwh6q
             pr3dp5ny3pox3axi/include -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/drivers/matfuncs -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/dri
             vers/matfuncs/scamac -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/drivers/matfuncs/scamac -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/
             spack-build-5bv4pof -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels/b
             uiltin -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels/trilinos_interface -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/too
             ls -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/core -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/krylov -I/tmp/root/spack-stag
             e/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/jada -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/precon -I/tmp/root/spack-stage/spack-stage-phist-1.11.
             2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/drivers -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof/cxx_bindings -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagc
             ksw3uh3wrx4lezb56myb/spack-src/lib/googletest-master-12-2015 -I/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src -march=native       -fopenmp -O3 -fno-math-errno -ffinite-math-only -fno
             -signed-zeros -fno-trapping-math -std=gnu99 -fPIC -MD -MT CMakeFiles/phist_kernels_builtin.dir/src/kernels/common/stream_bench.c.o -MF CMakeFiles/phist_kernels_builtin.dir/src/kernels/common/stream_bench.c.o.d -o CMakeFiles/phist
             _kernels_builtin.dir/src/kernels/common/stream_bench.c.o -c /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels/common/stream_bench.c
     1155    /spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/cmake-3.26.3-cib64fo6iqykcbtcbmtgdyivvlcof2s2/bin/cmake -E cmake_copy_f90_mod tritoeplitz_module.mod CMakeFiles/phist_kernels_builtin.dir/tritoeplitz_module.mod.stamp GNU
     1156    /spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/cmake-3.26.3-cib64fo6iqykcbtcbmtgdyivvlcof2s2/bin/cmake -E cmake_copy_f90_mod brussolator_module.mod CMakeFiles/phist_kernels_builtin.dir/brussolator_module.mod.stamp GNU
     1157    /spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/cmake-3.26.3-cib64fo6iqykcbtcbmtgdyivvlcof2s2/bin/cmake -E touch CMakeFiles/phist_kernels_builtin.dir/drivers/matfuncs/tritoeplitz.F90.o.provides.build
     1158    /spack/opt/spack/linux-ubuntu20.04-aarch64/gcc-11.1.0/cmake-3.26.3-cib64fo6iqykcbtcbmtgdyivvlcof2s2/bin/cmake -E touch CMakeFiles/phist_kernels_builtin.dir/drivers/matfuncs/brussolator.F90.o.provides.build
  >> 1159    /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels/builtin/axpy_kernels_nt.c:22:10: fatal error: emmintrin.h: No such file or directory
     1160       22 | #include <emmintrin.h>
     1161          |          ^~~~~~~~~~~~~
     1162    compilation terminated.
  >> 1163    /tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-src/src/kernels/builtin/spmvm_kernels_nt.c:22:10: fatal error: emmintrin.h: No such file or directory
     1164       22 | #include <emmintrin.h>
     1165          |          ^~~~~~~~~~~~~
     1166    compilation terminated.
  >> 1167    make[2]: *** [CMakeFiles/phist_kernels_builtin.dir/build.make:292: CMakeFiles/phist_kernels_builtin.dir/src/kernels/builtin/axpy_kernels_nt.c.o] Error 1
     1168    make[2]: *** Waiting for unfinished jobs....
  >> 1169    make[2]: *** [CMakeFiles/phist_kernels_builtin.dir/build.make:397: CMakeFiles/phist_kernels_builtin.dir/src/kernels/builtin/spmvm_kernels_nt.c.o] Error 1
     1170    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof'
  >> 1171    make[1]: *** [CMakeFiles/Makefile2:317: CMakeFiles/phist_kernels_builtin.dir/all] Error 2
     1172    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-phist-1.11.2-5bv4pofacagcksw3uh3wrx4lezb56myb/spack-build-5bv4pof'
  >> 1173    make: *** [Makefile:169: all] Error 2
```

### Error message

<details><summary>Error message</summary>
<pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.21.0.dev0 (09e1258ed42f00ea9bde9857f233eeda5c61a895)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-neoverse_n1
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/11992640/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/11992641/spack-build-out.txt)

@jthies @wspear 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/phist/avoid-sse.patch
+++ b/var/spack/repos/builtin/packages/phist/avoid-sse.patch
@@ -1,0 +1,189 @@
+commit bb27ecc296af03b174e758fa35e9ca3b96bc6756
+Author: Jonas Thies <16190001+jthies@users.noreply.github.com>
+Date:   Sun Jul 9 21:33:30 2023 +0200
+
+    exit early from builtin kernels requiring SSE so that they are not compiled if it is not available
+    (this broke phist compilation on ARM systems, even though we never called these kernels if SSE was disabled)
+
+diff --git a/src/kernels/builtin/axpy_kernels_nt.c b/src/kernels/builtin/axpy_kernels_nt.c
+index 64d5fbd0..f2c065ed 100644
+--- a/src/kernels/builtin/axpy_kernels_nt.c
++++ b/src/kernels/builtin/axpy_kernels_nt.c
+@@ -19,7 +19,9 @@
+ #endif
+ #include <stdint.h>
+ #include <stdio.h>
++#ifdef PHIST_HAVE_SSE
+ #include <emmintrin.h>
++#endif
+ #include <stdlib.h>
+ 
+ static inline _Bool is_aligned(const void *restrict pointer, size_t byte_count)
+@@ -30,6 +32,10 @@ static inline _Bool is_aligned(const void *restrict pointer, size_t byte_count)
+ 
+ void daxpy_nt_2_c(int nrows, const double *restrict alpha, const double *restrict x, double *restrict y)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(y,16) )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
+@@ -59,6 +65,10 @@ void daxpy_nt_2_c(int nrows, const double *restrict alpha, const double *restric
+ 
+ void daxpy_nt_4_c(int nrows, const double *restrict alpha, const double *restrict x, double *restrict y)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(y,16) )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
+@@ -91,6 +101,10 @@ void daxpy_nt_4_c(int nrows, const double *restrict alpha, const double *restric
+ 
+ void daxpy_nt_8_c(int nrows, const double *restrict alpha, const double *restrict x, double *restrict y)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(y,16) )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
+@@ -123,6 +137,10 @@ void daxpy_nt_8_c(int nrows, const double *restrict alpha, const double *restric
+ 
+ void daxpy_nt_strided_2_c(int nrows, const double *restrict alpha, const double *restrict x, int ldx, double *restrict y, int ldy)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(y,16) )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
+@@ -145,6 +163,10 @@ void daxpy_nt_strided_2_c(int nrows, const double *restrict alpha, const double
+ 
+ void daxpy_nt_strided_4_c(int nrows, const double *restrict alpha, const double *restrict x, int ldx, double *restrict y, int ldy)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(y,16) || ldy % 2 != 0 )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
+@@ -170,6 +192,10 @@ void daxpy_nt_strided_4_c(int nrows, const double *restrict alpha, const double
+ 
+ void daxpy_nt_strided_8_c(int nrows, const double *restrict alpha, const double *restrict x, int ldx, double *restrict y, int ldy)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(y,16) || ldy % 2 != 0 )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
+@@ -195,6 +221,10 @@ void daxpy_nt_strided_8_c(int nrows, const double *restrict alpha, const double
+ 
+ void dcopy_general_nt_c(int nrows, int nvec, const double *restrict x, int ldx, double *restrict y, int ldy)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( nvec % 2 != 0 )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)x);
+diff --git a/src/kernels/builtin/spmvm_kernels_nt.c b/src/kernels/builtin/spmvm_kernels_nt.c
+index d4d30bff..5a8db7c8 100644
+--- a/src/kernels/builtin/spmvm_kernels_nt.c
++++ b/src/kernels/builtin/spmvm_kernels_nt.c
+@@ -19,7 +19,9 @@
+ #endif
+ #include <stdint.h>
+ #include <stdio.h>
++#ifdef PHIST_HAVE_SSE
+ #include <emmintrin.h>
++#endif
+ #include <stdlib.h>
+ 
+ #ifdef PHIST_HIGH_PRECISION_KERNELS
+@@ -35,6 +37,10 @@ static inline _Bool is_aligned(const void *restrict pointer, size_t byte_count)
+ void dspmvm_nt_1_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
+                  const double *restrict shifts, const double *restrict rhsv, const double *restrict halo, double *restrict lhsv)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(lhsv,16) )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
+@@ -153,6 +159,10 @@ void dspmvm_nt_1_c(int nrows, double alpha, const long *restrict row_ptr, const
+ void dspmvm_nt_2_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
+                  const double *restrict shifts, const double *restrict rhsv, const double *restrict halo, double *restrict lhsv, int ldl)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(lhsv,32) || ldl % 2 != 0 )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
+@@ -242,6 +252,10 @@ void dspmvm_nt_2_c(int nrows, double alpha, const long *restrict row_ptr, const
+ void dspmvm_nt_4_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
+                  const double *restrict shifts, const double *restrict rhsv, const double *restrict halo, double *restrict lhsv, int ldl)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(lhsv,32) || ldl % 4 != 0 )
+   {
+     printf("%s: lhsv not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
+@@ -348,6 +362,10 @@ void dspmvm_nt_4_c(int nrows, double alpha, const long *restrict row_ptr, const
+ void dspmvm_nt_8_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
+                  const double *restrict shifts, const double *restrict rhsv, const double *restrict halo, double *restrict lhsv, int ldl)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(lhsv,16) || ldl % 2 != 0 )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
+@@ -418,6 +436,10 @@ void dspmvm_nt_8_c(int nrows, double alpha, const long *restrict row_ptr, const
+ void dspmvm_nt_strided_2_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
+                          const double *restrict shifts, const double *restrict rhsv, int ldr, const double *restrict halo, double *restrict lhsv, int ldl)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(lhsv,16) || ldl % 2 != 0 )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
+@@ -469,6 +491,10 @@ void dspmvm_nt_strided_2_c(int nrows, double alpha, const long *restrict row_ptr
+ void dspmvm_nt_strided_4_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
+                          const double *restrict shifts, const double *restrict rhsv, int ldr, const double *restrict halo, double *restrict lhsv, int ldl)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(lhsv,16) || ldl % 2 != 0 )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
+@@ -531,6 +557,10 @@ void dspmvm_nt_strided_4_c(int nrows, double alpha, const long *restrict row_ptr
+ void dspmvm_nt_strided_8_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
+                          const double *restrict shifts, const double *restrict rhsv, int ldr, const double *restrict halo, double *restrict lhsv, int ldl)
+ {
++#ifndef PHIST_HAVE_SSE
++  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
++  exit(1);
++#endif
+   if( !is_aligned(lhsv,16) || ldl % 2 != 0 )
+   {
+     printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);

--- a/var/spack/repos/builtin/packages/phist/avoid-sse.patch
+++ b/var/spack/repos/builtin/packages/phist/avoid-sse.patch
@@ -1,4 +1,4 @@
-commit bb27ecc296af03b174e758fa35e9ca3b96bc6756
+commit eaef462cc07509fe8f380fbf520a2617b910b139
 Author: Jonas Thies <16190001+jthies@users.noreply.github.com>
 Date:   Sun Jul 9 21:33:30 2023 +0200
 
@@ -6,7 +6,7 @@ Date:   Sun Jul 9 21:33:30 2023 +0200
     (this broke phist compilation on ARM systems, even though we never called these kernels if SSE was disabled)
 
 diff --git a/src/kernels/builtin/axpy_kernels_nt.c b/src/kernels/builtin/axpy_kernels_nt.c
-index 64d5fbd0..f2c065ed 100644
+index 64d5fbd0..17c5024a 100644
 --- a/src/kernels/builtin/axpy_kernels_nt.c
 +++ b/src/kernels/builtin/axpy_kernels_nt.c
 @@ -19,7 +19,9 @@
@@ -26,78 +26,121 @@ index 64d5fbd0..f2c065ed 100644
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(y,16) )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
-@@ -59,6 +65,10 @@ void daxpy_nt_2_c(int nrows, const double *restrict alpha, const double *restric
+@@ -54,11 +60,16 @@ void daxpy_nt_2_c(int nrows, const double *restrict alpha, const double *restric
+     // non-temporal store
+     _mm_stream_pd(y+2*i, y_);
+   }
++#endif
+ }
+ 
  
  void daxpy_nt_4_c(int nrows, const double *restrict alpha, const double *restrict x, double *restrict y)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(y,16) )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
-@@ -91,6 +101,10 @@ void daxpy_nt_4_c(int nrows, const double *restrict alpha, const double *restric
+@@ -86,11 +97,16 @@ void daxpy_nt_4_c(int nrows, const double *restrict alpha, const double *restric
+       _mm_stream_pd(y+4*i+2*k, y_);
+     }
+   }
++#endif
+ }
+ 
  
  void daxpy_nt_8_c(int nrows, const double *restrict alpha, const double *restrict x, double *restrict y)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(y,16) )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
-@@ -123,6 +137,10 @@ void daxpy_nt_8_c(int nrows, const double *restrict alpha, const double *restric
+@@ -118,11 +134,16 @@ void daxpy_nt_8_c(int nrows, const double *restrict alpha, const double *restric
+       _mm_stream_pd(y+8*i+2*k, y_);
+     }
+   }
++#endif
+ }
+ 
  
  void daxpy_nt_strided_2_c(int nrows, const double *restrict alpha, const double *restrict x, int ldx, double *restrict y, int ldy)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(y,16) )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
-@@ -145,6 +163,10 @@ void daxpy_nt_strided_2_c(int nrows, const double *restrict alpha, const double
+@@ -140,11 +161,16 @@ void daxpy_nt_strided_2_c(int nrows, const double *restrict alpha, const double
+     // non-temporal store
+     _mm_stream_pd(y+ldy*i, y_);
+   }
++#endif
+ }
+ 
  
  void daxpy_nt_strided_4_c(int nrows, const double *restrict alpha, const double *restrict x, int ldx, double *restrict y, int ldy)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(y,16) || ldy % 2 != 0 )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
-@@ -170,6 +192,10 @@ void daxpy_nt_strided_4_c(int nrows, const double *restrict alpha, const double
+@@ -165,11 +191,16 @@ void daxpy_nt_strided_4_c(int nrows, const double *restrict alpha, const double
+       _mm_stream_pd(y+ldy*i+2*k, y_);
+     }
+   }
++#endif
+ }
+ 
  
  void daxpy_nt_strided_8_c(int nrows, const double *restrict alpha, const double *restrict x, int ldx, double *restrict y, int ldy)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(y,16) || ldy % 2 != 0 )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)y);
-@@ -195,6 +221,10 @@ void daxpy_nt_strided_8_c(int nrows, const double *restrict alpha, const double
+@@ -190,11 +221,16 @@ void daxpy_nt_strided_8_c(int nrows, const double *restrict alpha, const double
+       _mm_stream_pd(y+ldy*i+2*k, y_);
+     }
+   }
++#endif
+ }
+ 
  
  void dcopy_general_nt_c(int nrows, int nvec, const double *restrict x, int ldx, double *restrict y, int ldy)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( nvec % 2 != 0 )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)x);
+@@ -217,5 +253,6 @@ void dcopy_general_nt_c(int nrows, int nvec, const double *restrict x, int ldx,
+       _mm_stream_pd(y+i*ldy+2*j, tmp);
+     }
+   }
++#endif
+ }
+ 
 diff --git a/src/kernels/builtin/spmvm_kernels_nt.c b/src/kernels/builtin/spmvm_kernels_nt.c
-index d4d30bff..5a8db7c8 100644
+index d4d30bff..5d858878 100644
 --- a/src/kernels/builtin/spmvm_kernels_nt.c
 +++ b/src/kernels/builtin/spmvm_kernels_nt.c
 @@ -19,7 +19,9 @@
@@ -117,73 +160,187 @@ index d4d30bff..5a8db7c8 100644
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(lhsv,16) )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
-@@ -153,6 +159,10 @@ void dspmvm_nt_1_c(int nrows, double alpha, const long *restrict row_ptr, const
+@@ -123,7 +129,7 @@ void dspmvm_nt_1_c(int nrows, double alpha, const long *restrict row_ptr, const
+ #endif
+ 
+   // last row
+-#ifdef PHIST_HIGH_PRECISION_KERNELS
++# ifdef PHIST_HIGH_PRECISION_KERNELS
+   if( nrows % 2 != 0 )
+   {
+     double lhs, lhsC;
+@@ -136,7 +142,7 @@ void dspmvm_nt_1_c(int nrows, double alpha, const long *restrict row_ptr, const
+ 
+     lhsv[nrows-1] = alpha*(lhs+lhsC);
+   }
+-#else
++# else
+   if( nrows % 2 != 0 )
+   {
+     lhsv[nrows-1] = shifts[0]*rhsv[nrows-1];
+@@ -146,6 +152,7 @@ void dspmvm_nt_1_c(int nrows, double alpha, const long *restrict row_ptr, const
+       lhsv[nrows-1] += val[j]*halo[ (col_idx[j]-1) ];
+     lhsv[nrows-1] *= alpha;
+   }
++# endif
+ #endif
+ }
+ 
+@@ -153,6 +160,10 @@ void dspmvm_nt_1_c(int nrows, double alpha, const long *restrict row_ptr, const
  void dspmvm_nt_2_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
                   const double *restrict shifts, const double *restrict rhsv, const double *restrict halo, double *restrict lhsv, int ldl)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(lhsv,32) || ldl % 2 != 0 )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
-@@ -242,6 +252,10 @@ void dspmvm_nt_2_c(int nrows, double alpha, const long *restrict row_ptr, const
+@@ -176,7 +187,7 @@ void dspmvm_nt_2_c(int nrows, double alpha, const long *restrict row_ptr, const
+   __m128d shifts_ = _mm_loadu_pd(shifts);
+   __m128d alpha_ = _mm_set1_pd(alpha);
+ 
+-#ifdef PHIST_HIGH_PRECISION_KERNELS
++# ifdef PHIST_HIGH_PRECISION_KERNELS
+ #pragma omp parallel for schedule(static)
+   for(int i = 0; i < nrows; i++)
+   {
+@@ -204,7 +215,7 @@ void dspmvm_nt_2_c(int nrows, double alpha, const long *restrict row_ptr, const
+     // non-temporal store
+     _mm_stream_pd(lhsv+i*ldl, lhs);
+   }
+-#else
++# else
+ #pragma omp parallel for schedule(static)
+   for(int i = 0; i < nrows; i++)
+   {
+@@ -232,16 +243,21 @@ void dspmvm_nt_2_c(int nrows, double alpha, const long *restrict row_ptr, const
+     // multiply with alpha
+     __m128d alpha_ = _mm_set1_pd(alpha);
+     lhs_ = _mm_mul_pd(alpha_,lhs_);
+- 
++
+     // non-temporal store
+     _mm_stream_pd(lhsv+i*ldl, lhs_);
+   }
++# endif
+ #endif
+ }
+ 
  void dspmvm_nt_4_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
                   const double *restrict shifts, const double *restrict rhsv, const double *restrict halo, double *restrict lhsv, int ldl)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(lhsv,32) || ldl % 4 != 0 )
    {
      printf("%s: lhsv not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
-@@ -348,6 +362,10 @@ void dspmvm_nt_4_c(int nrows, double alpha, const long *restrict row_ptr, const
+@@ -261,7 +277,7 @@ void dspmvm_nt_4_c(int nrows, double alpha, const long *restrict row_ptr, const
+   }
+ 
+ 
+-#ifdef PHIST_HIGH_PRECISION_KERNELS
++# ifdef PHIST_HIGH_PRECISION_KERNELS
+ 
+   __m256d shifts_ = _mm256_loadu_pd(shifts);
+   __m256d alpha_ = _mm256_set1_pd(alpha);
+@@ -294,7 +310,7 @@ void dspmvm_nt_4_c(int nrows, double alpha, const long *restrict row_ptr, const
+     _mm256_stream_pd(lhsv+i*ldl, lhs);
+   }
+ 
+-#else
++# else
+ 
+   __m128d shifts_[2];
+   shifts_[0] = _mm_loadu_pd(shifts);
+@@ -341,6 +357,7 @@ void dspmvm_nt_4_c(int nrows, double alpha, const long *restrict row_ptr, const
+     }
+   }
+ 
++# endif
+ #endif
+ }
+ 
+@@ -348,6 +365,10 @@ void dspmvm_nt_4_c(int nrows, double alpha, const long *restrict row_ptr, const
  void dspmvm_nt_8_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
                   const double *restrict shifts, const double *restrict rhsv, const double *restrict halo, double *restrict lhsv, int ldl)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(lhsv,16) || ldl % 2 != 0 )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
-@@ -418,6 +436,10 @@ void dspmvm_nt_8_c(int nrows, double alpha, const long *restrict row_ptr, const
+@@ -412,12 +433,17 @@ void dspmvm_nt_8_c(int nrows, double alpha, const long *restrict row_ptr, const
+       _mm_stream_pd(lhsv+i*ldl+2*k, lhs_[k]);
+     }
+   }
++#endif
+ }
+ 
+ 
  void dspmvm_nt_strided_2_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
                           const double *restrict shifts, const double *restrict rhsv, int ldr, const double *restrict halo, double *restrict lhsv, int ldl)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(lhsv,16) || ldl % 2 != 0 )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
-@@ -469,6 +491,10 @@ void dspmvm_nt_strided_2_c(int nrows, double alpha, const long *restrict row_ptr
+@@ -460,15 +486,20 @@ void dspmvm_nt_strided_2_c(int nrows, double alpha, const long *restrict row_ptr
+     // multiply with alpha
+     __m128d alpha_ = _mm_set1_pd(alpha);
+     lhs_ = _mm_mul_pd(alpha_,lhs_);
+- 
++
+     // non-temporal store
+     _mm_stream_pd(lhsv+i*ldl, lhs_);
+   }
++#endif
+ }
+ 
  void dspmvm_nt_strided_4_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
                           const double *restrict shifts, const double *restrict rhsv, int ldr, const double *restrict halo, double *restrict lhsv, int ldl)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(lhsv,16) || ldl % 2 != 0 )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
-@@ -531,6 +557,10 @@ void dspmvm_nt_strided_4_c(int nrows, double alpha, const long *restrict row_ptr
+@@ -526,11 +557,16 @@ void dspmvm_nt_strided_4_c(int nrows, double alpha, const long *restrict row_ptr
+       _mm_stream_pd(lhsv+i*ldl+2*k, lhs_[k]);
+     }
+   }
++#endif
+ }
+ 
  void dspmvm_nt_strided_8_c(int nrows, double alpha, const long *restrict row_ptr, const long *restrict halo_ptr, const int *restrict col_idx, const double *restrict val,
                           const double *restrict shifts, const double *restrict rhsv, int ldr, const double *restrict halo, double *restrict lhsv, int ldl)
  {
 +#ifndef PHIST_HAVE_SSE
 +  printf("%s: must not be called on platforms without SSE.", __FUNCTION__);
 +  exit(1);
-+#endif
++#else
    if( !is_aligned(lhsv,16) || ldl % 2 != 0 )
    {
      printf("%s: not aligned %lx\n", __FUNCTION__, (uintptr_t)(void*)lhsv);
+@@ -589,6 +625,7 @@ void dspmvm_nt_strided_8_c(int nrows, double alpha, const long *restrict row_ptr
+       _mm_stream_pd(lhsv+i*ldl+2*k, lhs_[k]);
+     }
+   }
++#endif
+ }
+ 
+ 

--- a/var/spack/repos/builtin/packages/phist/package.py
+++ b/var/spack/repos/builtin/packages/phist/package.py
@@ -150,6 +150,10 @@ class Phist(CMakePackage):
 
     # ###################### Patches ##########################
 
+    # Avoid trying to compile some SSE code if SSE is not available
+    # This patch will be part of phist 1.11.3 and greater and only affects
+    # the 'builtin' kernel_lib.
+    patch("avoid-sse.patch", when="@:1.11.2 kernel_lib=builtin")
     # Only applies to 1.9.4: While SSE instructions are handled correctly,
     # build fails on ppc64le unless -DNO_WARN_X86_INTRINSICS is defined.
     patch("ppc64_sse.patch", when="@1.9.4")

